openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/AbstractCoreRoutingAlgorithm.java

```diff
@@ -160,11 +172,42 @@ public RoutingAlgorithm setEdgeFilter(CoreDijkstraFilter additionalEdgeFilter) {
         this.additionalCoreEdgeFilter = additionalEdgeFilter;
         return this;
     }
-    protected boolean accept(EdgeIterator iter, int prevOrNextEdgeId) {
-        return additionalCoreEdgeFilter == null || additionalCoreEdgeFilter.accept(iter);
+
+    //TODO: refactor CoreEdgeFilter to plain EdgeFilter to avoid overriding this method
+    @Override
+    protected boolean accept(EdgeIteratorState iter, int prevOrNextEdgeId) {
+        if (iter.getEdge() == prevOrNextEdgeId) {
+            return false;
+        } else {
+            return additionalCoreEdgeFilter == null || additionalCoreEdgeFilter.accept(iter);
+        }
     }
 
     protected SPTEntry createSPTEntry(int node, double weight) {
         return new SPTEntry(EdgeIterator.NO_EDGE, node, weight);
     }
+
+    void updateBestPath(SPTEntry entryCurrent, SPTEntry entryOther, double newWeight, boolean reverse) {
+        bestPath.setSwitchToFrom(reverse);
+        bestPath.setSPTEntry(entryCurrent);
+        bestPath.setWeight(newWeight);
+        bestPath.setSPTEntryTo(entryOther);
+    }
+
+    boolean isCoreNode(int node) {
+        return chGraph.getLevel(node) >= coreNodeLevel;
+    }
+
+    boolean isTurnRestrictedNode(int node) {
+        return chGraph.getLevel(node) == turnRestrictedNodeLevel;
+    }
+
+    boolean considerTurnRestrictions(int node) {
+        if (!hasTurnWeighting)
+            return false;
+        if (approximate)
```

Comment:
It is currently not possible to use the approximated algorithm other than by setting the corresponding flag in the source code. This is on purpose so and there are various reasons for it. First of all, once the approximation is enabled the resulting route for a given weighting is no longer guaranteed to be the optimal one. This is different from the rest of the algorithms we use which are exact in those terms. The reason for this is mostly because alternative paths involving edges which have been reached previously on a shorter way are not explored anymore. This can happen, for example,  when the forward or backward search enters a street from both ends. Once a node on such street is relaxed it would not be considered anymore, even though because of turn restrictions the longer path going in the opposite direction might actually be eventually involved in the shortest path in question. This is solved by the GH's original edge-based algorithm by distinguishing between the direction of the edges. In order to reproduce this in node-based routing one needs to keep a list of weights corresponding to the incoming edges for each and every node. The approximated algorithm does this only for nodes involved in turn relations. Although some initial testing suggests that the resulting differences tend to be small and occur in < 1% of the routes, it merits a more thorough assessment in order to better understand its practical implications. But more importantly, while performance-wise it makes quite a difference for CoreDijkstra, I was not able to observe any significant impact on CoreALT, at least on the Berlin dataset. There was virtually no measurable difference in query times between the exact and the approximate solution (I shall upload the results on Confluence). Therefore, I've decided to keep the implementation dormant with the option to easily switch it on in the source code in case we are interested in pursuing this idea in the future.

Reply:
Thanks for the explanation, sounds reasonable to me given that the performance with CALT is no different.